src/main/kotlin/com/terraformation/backend/accelerator/db/CohortModuleStore.kt

```diff
@@ -0,0 +1,146 @@
+package com.terraformation.backend.accelerator.db
+
+import com.terraformation.backend.accelerator.model.ModuleModel
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.customer.model.requirePermissions
+import com.terraformation.backend.db.accelerator.CohortId
+import com.terraformation.backend.db.accelerator.ModuleId
+import com.terraformation.backend.db.accelerator.ParticipantId
+import com.terraformation.backend.db.accelerator.tables.references.COHORT_MODULES
+import com.terraformation.backend.db.accelerator.tables.references.MODULES
+import com.terraformation.backend.db.accelerator.tables.references.PARTICIPANTS
+import com.terraformation.backend.db.default_schema.ProjectId
+import com.terraformation.backend.db.default_schema.tables.references.PROJECTS
+import jakarta.inject.Named
+import java.time.LocalDate
+import org.jooq.Condition
+import org.jooq.DSLContext
+import org.jooq.Record
+import org.jooq.SelectOnConditionStep
+import org.jooq.impl.DSL
+
+@Named
+class CohortModuleStore(
+    private val dslContext: DSLContext,
+) {
+  fun fetch(
+      cohortId: CohortId? = null,
+      participantId: ParticipantId? = null,
+      projectId: ProjectId? = null,
+      moduleId: ModuleId? = null,
+  ): List<ModuleModel> {
+    requirePermissions {
+      when {
+        projectId != null -> readProjectModules(projectId)
+        participantId != null -> readParticipant(participantId)
+        cohortId != null -> readCohort(cohortId)
+        else -> manageModules()
+      }
+
+      moduleId?.let { readModule(it) }
+    }
+
+    val condition =
+        DSL.and(
+            when {
+              projectId != null -> PROJECTS.ID.eq(projectId)
+              participantId != null -> PARTICIPANTS.ID.eq(participantId)
+              cohortId != null -> COHORT_MODULES.COHORT_ID.eq(cohortId)
+              else -> null
+            },
+            moduleId?.let { MODULES.ID.eq(moduleId) })
+
+    val joinForVisibility = { query: SelectOnConditionStep<Record> ->
+      when {
+        projectId != null ->
+            query
+                .join(PARTICIPANTS)
+                .on(PARTICIPANTS.COHORT_ID.eq(COHORT_MODULES.COHORT_ID))
+                .join(PROJECTS)
+                .on(PROJECTS.PARTICIPANT_ID.eq(PARTICIPANTS.ID))
+        participantId != null ->
+            query.join(PARTICIPANTS).on(PARTICIPANTS.COHORT_ID.eq(COHORT_MODULES.COHORT_ID))
+        else -> query
+      }
+    }
+
+    return fetch(condition, joinForVisibility)
+  }
+
+  fun assign(
```

Comment:
Assign/Remove functionalities are added here to replace handling those as part of CohortStore, update functions.